### PR TITLE
CURA-4234 Make sidebar collapsible 

### PR DESCRIPTION
**Pull request by jackha**
CURA-4234

Ctrl-E or using the View pull down menu

### CODE REVIEW

File: resources/qml/Cura.qml [hunk 1]
```diff
@@ -49,6 +67,15 @@ UM.MainWindow
         // This has been fixed for QtQuick Controls 2 since the Shortcut item has a context property.
         Cura.Actions.parent = backgroundItem
         CuraApplication.purgeWindows()
+
+
+        var sidebarCollaps = UM.Preferences.getValue("general/sidebar_collaps")
```
Comment (jackha) [inline resources/qml/Cura.qml]: Correct spelling "sidebar_collapse"

File: resources/qml/Cura.qml [hunk 2]
```diff
@@ -369,16 +387,59 @@ UM.MainWindow
...
+                    id: expandSidebarAnimation
+                    target: sidebar
+                    properties: "x"
+                    to: base.width - sidebar.width
+                    duration: 500
+                }
+
+                Component.onCompleted:
+                {
+                    var sidebarCollaps = UM.Preferences.getValue("general/sidebar_collaps")
```
Comment (jackha) [inline resources/qml/Cura.qml]: The correct spelling is "collapse".

File: resources/qml/Cura.qml [hunk 3]
```diff
@@ -369,16 +387,59 @@ UM.MainWindow
...
+                    to: base.width
+                    duration: 500
+                }
+
+                NumberAnimation {
+                    id: expandSidebarAnimation
+                    target: sidebar
+                    properties: "x"
+                    to: base.width - sidebar.width
+                    duration: 500
```
Comment (jackha) [inline resources/qml/Cura.qml]: Let's make the duration either very fast (100ms) or remove the animation all together.

File: resources/qml/Cura.qml (hunk 2 above)
Comment (alekseisasin) [reply resources/qml/Cura.qml]: ok

**Comment by Appesteijn** [comment]
Component tested on MacOS == OK

**Pull request merged**